hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/PayloadCreation.java

```diff
@@ -43,14 +44,17 @@ public class PayloadCreation implements Serializable {
   private static final long serialVersionUID = 1L;
 
   private final boolean shouldCombine;
+  private final boolean shouldUsePropsForPayload;
   private final Constructor<?> constructor;
   private final String preCombineField;
 
   private PayloadCreation(
       boolean shouldCombine,
+      boolean shouldUsePropsForPayload,
       Constructor<?> constructor,
       @Nullable String preCombineField) {
     this.shouldCombine = shouldCombine;
+    this.shouldUsePropsForPayload = shouldUsePropsForPayload;
```

Comment:
`shouldUsePropsForPayload` should be always true?

Reply:
No, for the record payload class implemented by a user outside Hudi repo, the class may have the constructor with old signature, i.e., `{GenericRecord.class, Comparable.class}` or `{Option.class}`, so to be compatible on that, there's a fallback mechanism to create payload without properties (which is OK is the payload does not leverage any props).  See `PayloadCreation#instance`.